### ISSUE TITLE
hacs-isy994 v1.3.4 - ignored entities

### ISSUE THREAD

**Issue by timwing** [shbatm/hacs-isy994#43]
@shbatm got more errors after upgrading to v1.3.4.

```
2020-04-11 22:23:18 ERROR (SyncWorker_9) [custom_components.isy994] ISY Could not parse response, poorly formatted XML.: NetworkResources
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.lock] Entity id already exists - ignoring: lock.front_door_lock. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.master_bedroom_lights. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.dining_room_lights. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.kitchen_lights. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.zw_002_multilevel_switch. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.exterior_lights_front. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.living_room_lights_front. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.stairway_lights_down. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.stairway_lights_up. Platform isy994 does not generate unique IDs
2020-04-11 22:23:20 ERROR (MainThread) [homeassistant.components.light] Entity id already exists - ignoring: light.living_room_lights_back. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.driveway_detection_relay. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.reloading_room_outlet. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.reloading_room_light. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.garage_lights. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.garage_door_house_relay. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.exterior_lights_all. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.garage_door_warning_lights. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.auto_dr. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.bedtime_lights_out. Platform isy994 does not generate unique IDs
2020-04-11 22:23:21 ERROR (MainThread) [homeassistant.components.switch] Entity id already exists - ignoring: switch.kodi_event. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.driveway_detection_status. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.zw_002_color_switch. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.garage_door_house_off_closed. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.myq_service. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.house_garage_door. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.shop_garage_door. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.forecast_5. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.forecast_6. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.zone2. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.zone3. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.bond_bridge. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.blue_iris. Platform isy994 does not generate unique IDs
2020-04-11 22:23:22 ERROR (MainThread) [homeassistant.components.binary_sensor] Entity id already exists - ignoring: binary_sensor.zw_003_color_switch. Platform isy994 does not generate unique IDs
```

**Comment by shbatm** [comment]
Do you have the ability to test on a completely fresh install of Home Assistant? For some reason you still have latent entities stuck in memory and I don't know what else to try except testing on a fresh install, or deleting the following files from config/.storage (while HA is stopped)

- core.entity_registry
- core.device_registry
- core.restore_state

**Comment by timwing** [comment]
Ok @shbatm I have uninstalled hacs-isy994 from hacs & components, deleted the three files mentioned in the .storage folder. They are recreated upon start of HA. I searched each file and the only reference to "isy" are the 3 unifi entries in the core.entity_registry file. Should be clean, yes?

Please let me know how to proceed from here. I assume the following but will wait for your confirmation.

1. Install hacs-isy994 v1.3.4 from HACS
2. Integrations --> Setup new integration --> ISY
3. Add my first ISY then repeat for the second ISY.

Thank you

**Comment by shbatm** [comment]
Yep. Add one at a time, and make sure you don't have anything defined in `configuration.yaml` to make sure you don't have any duplicates.

**Comment by timwing** [comment]
Here is my log before hacs-isy994 v1.3.4...

```
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for bond which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for anniversaries which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for sonoff which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for nodered which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for browser_mod which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for nexia which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for blueiris which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:51:44 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 12:52:09 WARNING (MainThread) [homeassistant.components.switch] Setup of switch platform alexa_media is taking over 10 seconds.
2020-04-12 12:52:09 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform alexa_media is taking over 10 seconds.
```


and after....

```
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for nodered which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for bond which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for browser_mod which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for anniversaries which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for nexia which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for isy994 which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for sonoff which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:48 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for blueiris which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-04-12 13:17:56 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry ISY - House (192.168.0.3) for isy994
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 216, in async_setup
    hass, self
  File "/config/custom_components/isy994/__init__.py", line 510, in async_setup_entry
    webroot=host.path,
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.7/site-packages/pyisy/isy.py", line 86, in __init__
    self.programs = Programs(self, xml=self.conn.get_programs())
  File "/usr/local/lib/python3.7/site-packages/pyisy/programs/__init__.py", line 97, in __init__
    self.parse(xml)
  File "/usr/local/lib/python3.7/site-packages/pyisy/programs/__init__.py", line 272, in parse
    pobj.update(data=data)
  File "/usr/local/lib/python3.7/site-packages/pyisy/programs/program.py", line 166, in update
    self.enabled = data["penabled"]
KeyError: 'penabled'
```

And after several restarts, I show no devices for 192.168.0.3 (House).

![image](https://user-images.githubusercontent.com/60765858/79075397-94c93200-7cc0-11ea-90ba-f9519afc4b5d.png)

**Comment by shbatm** [comment]
Please try to install a custom version of PyISY and post the log for the error again. You should have a line that looks like "Updating program ### with data ..." above the error.

To install, assuming you're using Docker/Hass.io:

```shell
docker exec -it homeassistant pip install -e git+https://github.com/shbatm/PyISY.git@prog_debug#egg=pyisy-beta==2.0.0rc3
```

Then restart Home Assistant

**Comment by timwing** [comment]
I am using HA 0.108.3 on a Raspberry Pi 4. Does that change anything?

Here is the xml.
```
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<programs>
<program id="0001" status="true" folder="true">
<name>Programs</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0054" parentId="0005" status="true" folder="true">
<name>01 - Times</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0055" parentId="0005" status="true" folder="true">
<name>02 - Temps</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0018" parentId="0001" status="true" folder="true">
<name>3-Way Switches</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="000D" parentId="0001" status="true" folder="true">
<name>DayTime</name>
<lastRunTime/>
<lastFinishTime/>
<nextScheduledRunTime>2020/04/12 7:50:51 PM</nextScheduledRunTime>
</program>
<program id="0059" parentId="0001" status="true" folder="true">
<name>Driveway</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0030" parentId="0001" status="true" folder="true">
<name>Master Bedroom</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0009" parentId="0001" status="true" folder="true">
<name>Media Center</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="000E" parentId="0001" status="false" folder="true">
<name>NightTime</name>
<lastRunTime/>
<lastFinishTime/>
<nextScheduledRunTime>2020/04/12 7:35:51 PM</nextScheduledRunTime>
</program>
<program id="0063" parentId="0001" status="true" folder="true">
<name>Overhead Garage Door</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0005" parentId="0001" status="true" folder="true">
<name>Thermostat</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0015" parentId="0001" status="true" folder="true">
<name>Timers</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="005C" parentId="0001" status="true" folder="true">
<name>Toyota Tundra</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0052" parentId="0001" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Entry Door Lock House - Low Battery</name>
<lastRunTime>2020/04/12 11:31:18 AM</lastRunTime>
<lastFinishTime>2020/04/12 11:31:18 AM</lastFinishTime>
</program>
<program id="0003" parentId="0001" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Reloading Room Lights</name>
<lastRunTime>2020/04/12 10:48:32 AM</lastRunTime>
<lastFinishTime>2020/04/12 10:48:32 AM</lastFinishTime>
</program>
<program id="0046" parentId="0001" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Lock Front Door</name>
<lastRunTime>2020/04/12 10:02:16 AM</lastRunTime>
<lastFinishTime>2020/04/12 10:02:16 AM</lastFinishTime>
<nextScheduledRunTime>2020/04/12 10:00:00 PM</nextScheduledRunTime>
</program>
<program id="0050" parentId="0030" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Sleeping Hours</name>
<lastRunTime>2020/04/12 9:00:01 AM</lastRunTime>
<lastFinishTime>2020/04/12 9:00:01 AM</lastFinishTime>
<nextScheduledRunTime>2020/04/12 9:00:00 PM</nextScheduledRunTime>
</program>
<program id="0056" parentId="0054" status="true" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>.Event_Weekend</name>
<lastRunTime>2020/04/12 7:00:00 AM</lastRunTime>
<lastFinishTime>2020/04/12 7:00:00 AM</lastFinishTime>
<nextScheduledRunTime>2020/04/12 9:00:00 PM</nextScheduledRunTime>
</program>
<program id="003E" parentId="0030" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Warning-Garage Door Open</name>
<lastRunTime>2020/04/12 4:00:01 AM</lastRunTime>
<lastFinishTime>2020/04/12 4:00:01 AM</lastFinishTime>
<nextScheduledRunTime>2020/04/12 9:30:00 PM</nextScheduledRunTime>
</program>
<program id="0002" parentId="0001" status="true" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Query All</name>
<lastRunTime>2020/04/12 3:00:00 AM</lastRunTime>
<lastFinishTime>2020/04/12 3:00:00 AM</lastFinishTime>
<nextScheduledRunTime>2020/04/13 3:00:00 AM</nextScheduledRunTime>
</program>
<program id="001D" parentId="0018" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Living Room 3-Way Back</name>
<lastRunTime>2020/04/12 12:58:23 AM</lastRunTime>
<lastFinishTime>2020/04/12 12:58:23 AM</lastFinishTime>
</program>
<program id="0010" parentId="0015" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Timer Heather Bathroom</name>
<lastRunTime>2020/04/12 12:58:23 AM</lastRunTime>
<lastFinishTime>2020/04/12 1:58:28 AM</lastFinishTime>
</program>
<program id="001B" parentId="0018" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Living Room 3-Way Front</name>
<lastRunTime>2020/04/12 12:58:22 AM</lastRunTime>
<lastFinishTime>2020/04/12 12:58:22 AM</lastFinishTime>
</program>
<program id="001E" parentId="000E" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Exterior Lights - Door Lock</name>
<lastRunTime>2020/04/12 12:56:24 AM</lastRunTime>
<lastFinishTime>2020/04/12 12:56:54 AM</lastFinishTime>
</program>
<program id="0037" parentId="0054" status="true" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>.Event_Week_Sleep</name>
<lastRunTime>2020/04/11 9:00:00 PM</lastRunTime>
<lastFinishTime>2020/04/11 9:00:00 PM</lastFinishTime>
<nextScheduledRunTime>2020/04/12 9:00:00 PM</nextScheduledRunTime>
</program>
<program id="0017" parentId="0063" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Garage Door House Status</name>
<lastRunTime>2020/04/11 3:04:07 PM</lastRunTime>
<lastFinishTime>2020/04/11 3:04:20 PM</lastFinishTime>
</program>
<program id="0016" parentId="0018" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Stairway 3-Way Down</name>
<lastRunTime>2020/04/11 3:04:07 PM</lastRunTime>
<lastFinishTime>2020/04/11 3:04:20 PM</lastFinishTime>
</program>
<program id="000C" parentId="0018" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Stairway 3-Way Up</name>
<lastRunTime>2020/04/11 3:04:07 PM</lastRunTime>
<lastFinishTime>2020/04/11 3:04:20 PM</lastFinishTime>
</program>
<program id="005E" parentId="0059" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Driveway Sensor Activated</name>
<lastRunTime>2020/04/11 3:04:06 PM</lastRunTime>
<lastFinishTime>2020/04/11 3:04:20 PM</lastFinishTime>
</program>
<program id="003B" parentId="0054" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>.Event_Week_Wake</name>
<lastRunTime/>
<lastFinishTime/>
<nextScheduledRunTime>2020/04/13 5:00:00 AM</nextScheduledRunTime>
</program>
<program id="003C" parentId="0054" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>.Event_Work-Leave</name>
<lastRunTime/>
<lastFinishTime/>
<nextScheduledRunTime>2020/04/13 7:00:00 AM</nextScheduledRunTime>
</program>
<program id="003D" parentId="0054" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>.Event_Work-Return</name>
<lastRunTime/>
<lastFinishTime/>
<nextScheduledRunTime>2020/04/13 4:45:00 PM</nextScheduledRunTime>
</program>
<program id="000B" parentId="000E" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Exterior Front Lights - Driveway</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0004" parentId="000D" status="true" folder="false" enabled="false" runAtStartup="false" running="idle">
<name>Exterior Lights OFF</name>
<lastRunTime/>
<lastFinishTime/>
<nextScheduledRunTime>2020/04/12 7:50:51 PM</nextScheduledRunTime>
</program>
<program id="0011" parentId="0063" status="false" folder="false" enabled="false" runAtStartup="false" running="idle">
<name>Garage Door House Notification</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0035" parentId="0063" status="false" folder="false" enabled="false" runAtStartup="false" running="idle">
<name>Garage Door House Status Check</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="001A" parentId="000D" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Interior Lights OFF</name>
<lastRunTime/>
<lastFinishTime/>
<nextScheduledRunTime>2020/04/13 7:04:01 AM</nextScheduledRunTime>
</program>
<program id="0022" parentId="0030" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>MBR Music OFF</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0021" parentId="0030" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>MBR Music ON</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="000F" parentId="0001" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>PANIC</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="005A" parentId="005C" status="false" folder="false" enabled="false" runAtStartup="false" running="idle">
<name>Tim - Lock Doors</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0026" parentId="005C" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Tim - Start Vehicle</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0019" parentId="005C" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Tim - Un Lock All Doors</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="005B" parentId="005C" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Tim - Un Lock Drivers Door</name>
<lastRunTime/>
<lastFinishTime/>
</program>
<program id="0006" parentId="000E" status="false" folder="false" enabled="true" runAtStartup="false" running="idle">
<name>Welcome Home</name>
<lastRunTime/>
<lastFinishTime/>
</program>
</programs>
```

**Comment by timwing** [comment]
> Please try to install a custom version of PyISY and post the log for the error again.

@shbatm do you want me to install the HACS version again?

**Comment by shbatm** [comment]
Yes: if you're not already using the HACS version, install it and restart HA once to make sure it installs correctly. Then run the command above and restart again. This will make sure you're on the custom version of PyISY to give the debug log.

**Comment by timwing** [comment]
pardon my ignorance here but where would I run this command in hass.io?

**Comment by shbatm** [comment]
It would be on the command line (ssh) for the host computer running hass.io. But disregard, I think I found the issue that was causing the error. It will be fixed in automicus/PyISY#92 and I'll push another update here shortly.
